tools/progen_compile.py

```diff
@@ -88,16 +90,26 @@ def build_projects(file, projects, tool, build=True, clean=False, ignore_errors=
                     nargs='*', type=str, default=[])
 parser.add_argument('--release', dest='release', action='store_true', help='Create a release with the yaml version file')
 parser.add_argument('--release-folder', type=str, default='firmware', help='Directory to create and place files in')
-parser.add_argument('--toolchain', type=str, default='GCC', help='Toolchain ("GCC" or "ARM", default: %(default)s',
-                    choices=['GCC', 'ARM'])
+parser.add_argument('--toolchain', type=str, default='GCC', help='Toolchain ("GCC", "ARMCC", "ARMCLANG", default: %(default)s',
+                    choices=['GCC', 'ARMCC', "ARMCLANG"])
 parser.add_argument('--clean', dest='clean', action='store_true', help='Rebuild or delete build folder before compile')
+parser.add_argument('--ignore-failures', dest='ignore_failures', action='store_true', help='Continue build even in case of failures')
+parser.add_argument('--parallel', dest='parallel', action='store_true', help='Build with multiple compilations in parallel')
 parser.add_argument('-v', dest='verbosity', action='count', help='Increase verbosity (can be specify multiple times)', default=0)
 parser.set_defaults(build=True)
 parser.set_defaults(clean=False)
+parser.set_defaults(ignore_failures=False)
+parser.set_defaults(parallel=False)
 parser.set_defaults(release=False)
 args = parser.parse_args()
 
-toolchain = "make_gcc_arm" if args.toolchain in ['GCC'] else 'make_armcc'
+toolchains = {
+    "GCC": "make_gcc_arm",
+    "ARMCC": "make_armcc",
+    "ARMCLANG": "make_armclang",
+}
+
+toolchain = toolchains[args.toolchain] if args.toolchain in toolchains.keys() else 'make_gcc_arm'
```

Comment:
You don't need the `in` test because the arg uses a `choices` list that is checked by argparse.

Anyway, a slightly cleaner way to write it would be to use the `get()` method:

```py
toolchain = toolchains.get(args.toolchain, 'make_gcc_arm')
```